Exception: Cannot add task 'testNativeImage'

### Expected Behavior

No exception

### Actual Behaviour

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project 'EXAMPLE'.
> Cannot add task 'testNativeImage' as a task with that name already exists.

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.ProjectConfigurationException: A problem occurred configuring project 'EXAMPLE'.
	at org.gradle.configuration.project.LifecycleProjectEvaluator.wrapException(LifecycleProjectEvaluator.java:75)
	at org.gradle.configuration.project.LifecycleProjectEvaluator.addConfigurationFailure(LifecycleProjectEvaluator.java:68)
	...
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:61)
Caused by: org.gradle.api.internal.tasks.DefaultTaskContainer$DuplicateTaskException: Cannot add task 'testNativeImage' as a task with that name already exists.
	at org.gradle.api.internal.tasks.DefaultTaskContainer.failOnDuplicateTask(DefaultTaskContainer.java:257)
	at org.gradle.api.internal.tasks.DefaultTaskContainer.registerTask(DefaultTaskContainer.java:398)
	at org.gradle.api.internal.tasks.DefaultTaskContainer.register(DefaultTaskContainer.java:375)
	at io.micronaut.gradle.graalvm.MicronautGraalPlugin.registerTestAgainstNativeImageTask(MicronautGraalPlugin.java:84)
	at io.micronaut.gradle.graalvm.MicronautGraalPlugin.lambda$null$6(MicronautGraalPlugin.java:73)
	at io.micronaut.gradle.graalvm.MicronautGraalPlugin.lambda$null$7(MicronautGraalPlugin.java:71)


### Steps To Reproduce

A complex project with the following in it's root `build.gradle.kts`:

```kotlin
subprojects {
  tasks {
        withType<Test> {
            failFast = true
            testLogging {
                events = setOf(org.gradle.api.tasks.testing.logging.TestLogEvent.FAILED)
                // NOTE: Uncomment this when you need a lot of output:
                // showStandardStreams = true
                exceptionFormat = org.gradle.api.tasks.testing.logging.TestExceptionFormat.FULL
            }
        }
  }
}
```

### Environment Information

_No response_

### Example Application

_No response_

### Version

3.1.1

--- a/src/main/java/io/micronaut/gradle/graalvm/MicronautGraalPlugin.java
+++ b/src/main/java/io/micronaut/gradle/graalvm/MicronautGraalPlugin.java
@@ -68,9 +68,10 @@ public class MicronautGraalPlugin implements Plugin<Project> {
 
             // We use `afterEvaluate` here in order to preserve laziness of task configuration
             // and because there is no API to allow reacting to registration of tasks.
+            Set<String> alreadyRegisteredTaskNames = new HashSet<>();
             project.afterEvaluate(p -> tasks.withType(Test.class).getCollectionSchema().getElements().forEach(element -> {
                 String testName = element.getName();
-                registerTestAgainstNativeImageTask(project, tasks, testName);
+                registerTestAgainstNativeImageTask(alreadyRegisteredTaskNames, tasks, testName);
             }));
         });
     }
@@ -80,30 +81,32 @@ public class MicronautGraalPlugin implements Plugin<Project> {
      * server. Note that this is different from the `nativeTest` task that the GraalVM Gradle plugin provides,
      * as the latter executes all tests _within_ the native image.
      */
-    private void registerTestAgainstNativeImageTask(Project project, TaskContainer tasks, String testName) {
-        tasks.register(testName + "NativeImage", Test.class, nativeImageTestTask -> {
-            Test testTask = (Test) tasks.getByName(testName);
-            nativeImageTestTask.setClasspath(testTask.getClasspath());
-            nativeImageTestTask.getJavaLauncher().set(testTask.getJavaLauncher());
-            BuildNativeImageTask nativeBuild = (BuildNativeImageTask) tasks.findByName("nativeCompile");
-            nativeImageTestTask.setForkEvery(testTask.getForkEvery());
-            nativeImageTestTask.setTestClassesDirs(testTask.getTestClassesDirs());
-            nativeImageTestTask.getJvmArgumentProviders().add(new CommandLineArgumentProvider() {
-                @InputFile
-                @PathSensitive(PathSensitivity.RELATIVE)
-                Provider<RegularFile> getInputFile() {
-                    return nativeBuild.getOutputFile();
-                }
+    private void registerTestAgainstNativeImageTask(Set<String> alreadyRegistered, TaskContainer tasks, String testName) {
+        if (alreadyRegistered.add(testName)) {
+            tasks.register(testName + "NativeImage", Test.class, nativeImageTestTask -> {
+                Test testTask = (Test) tasks.getByName(testName);
+                nativeImageTestTask.setClasspath(testTask.getClasspath());
+                nativeImageTestTask.getJavaLauncher().set(testTask.getJavaLauncher());
+                BuildNativeImageTask nativeBuild = (BuildNativeImageTask) tasks.findByName("nativeCompile");
+                nativeImageTestTask.setForkEvery(testTask.getForkEvery());
+                nativeImageTestTask.setTestClassesDirs(testTask.getTestClassesDirs());
+                nativeImageTestTask.getJvmArgumentProviders().add(new CommandLineArgumentProvider() {
+                    @InputFile
+                    @PathSensitive(PathSensitivity.RELATIVE)
+                    Provider<RegularFile> getInputFile() {
+                        return nativeBuild.getOutputFile();
+                    }
 
-                @Override
-                public Iterable<String> asArguments() {
-                    return Collections.singleton(
-                            "-Dmicronaut.test.server.executable=" + getInputFile().get().getAsFile().getAbsolutePath()
-                    );
-                }
+                    @Override
+                    public Iterable<String> asArguments() {
+                        return Collections.singleton(
+                                "-Dmicronaut.test.server.executable=" + getInputFile().get().getAsFile().getAbsolutePath()
+                        );
+                    }
+                });
+                nativeImageTestTask.setDescription("Runs tests against a native image build of the server. Requires the server to allow the port to configurable with 'micronaut.server.port'.");
             });
-            nativeImageTestTask.setDescription("Runs tests against a native image build of the server. Requires the server to allow the port to configurable with 'micronaut.server.port'.");
-        });
+        }
     }
 
     private static void configureMicronautLibrary(Project project, MicronautExtension extension) {

